modules/core/src/main/scala/org/scalasteward/core/coursier/CoursierAlg.scala

```diff
@@ -126,15 +126,35 @@ object CoursierAlg {
 
   private def toCoursierRepository(resolver: Resolver): Either[String, coursier.Repository] =
     resolver match {
-      case Resolver.MavenRepository(_, location, creds) =>
-        Right(coursier.maven.MavenRepository.apply(location, creds.map(toCoursierAuthentication)))
-      case Resolver.IvyRepository(_, pattern, creds) =>
+      case Resolver.MavenRepository(_, location, creds, headers) =>
+        Right(
+          coursier.maven.MavenRepository
+            .apply(location, toCoursierAuthentication(creds, headers))
+        )
+      case Resolver.IvyRepository(_, pattern, creds, headers) =>
         coursier.ivy.IvyRepository
-          .parse(pattern, authentication = creds.map(toCoursierAuthentication))
+          .parse(pattern, authentication = toCoursierAuthentication(creds, headers))
     }
 
-  private def toCoursierAuthentication(credentials: Credentials): Authentication =
-    Authentication(credentials.user, credentials.pass)
+  private def toCoursierAuthentication(
+      credentials: Option[Credentials],
+      headers: List[Resolver.Header]
+  ): Option[Authentication] =
+    if (credentials.isEmpty && headers.isEmpty) {
+      None
+    } else {
+      Some(
+        new Authentication(
+          credentials.fold("")(_.user),
```

Comment:
Q) Does `""` for user works ?

Reply:
Yes, that what's Coursier also does internally when just specifying headers. See: https://github.com/coursier/coursier/blob/2cb552ea934a100f52ce79f94278304c90d808a4/modules/util/shared/src/main/scala/coursier/core/Authentication.scala#L88